objects/SetupChecker/script.lua

```diff
@@ -1955,19 +1955,19 @@ function getNewSpirit(tags, complexities)
 end
 function pickSpirit1(obj, color)
     if Global.getTable("playerTables")[color] ~= obj then return end
-    pickSpirit(obj, 3, color)
+    pickSpirit(obj, 1, color)
```

Comment:
should these be 0, 1, 2, 3 since you're removing all 3 bottom row buttons right?

Reply:
No. Lua arrays are 1 indexed. The reason this value was 3 instead of 4 previously is because `pickSpirit` was adding one to the value passed to it, which I've removed here.

Reply:
tts buttons are 0 indexed, so you'll either want to leave the offsetting how it was before, or pass in index - 1 into the replaceSpirit call

Reply:
However, the entries returned by `.getButtons()` is a lua array that is 1 indexed (but the tables representing each button have an `index` entry that has the 0-indexed value that is 1 less than the lua array index for each.

Reply:
okay let me try saying this differently, there is a bug with how you're handling indexes currently so you can either:

- have the indexes be 0,1,2,3 and put the +1 back
- or pass in index - 1 into the replaceSpirit call

Reply:
Ah. I had missed the call to `replaceSpirit` there. I've switched this to pass in the TTS index now.

While I was looking at the code, I found a couple of other places that still needed adjusting to deal with the changed indices and fixed them.

I also discovered a bug with pressing "Gain Spirits" after `replaceSpirit` couldn't find a spirit to gain, and fixed that as well.

All that said, I suspect this code would be somewhat clearer if it was switch to use Custom UI as well, so I'll take a look at doing that after this lands.